Whenever the systems performs an operation that leads to changes (insertions or updates) over the data stored in \emph{Neo4j} we handle possible exceptions, testing for errors in the execution of the code as follows: if an error occurs in the \emph{try} block, we wait for a certain amount of time and we try to re-execute the operations (into the \emph{catch} statement) until it is carried out correctly. 
We do this up to a defined maximum number of times $N = 10$, and at each try the time that the system will wait to make the next attempt will increase. Hopefully, by doing this, even when the server is down the operations will still be carried out once the server will get back up. If $N$ or the chosen time in between every try,  turn out to be not sufficient for the operation to be carried out correctly, than an error will be reported in the \emph{log} file.